link/src/main/java/com/stripe/android/link/serialization/PopupPayload.kt

```diff
@@ -0,0 +1,139 @@
+package com.stripe.android.link.serialization
+
+import android.content.Context
+import android.os.Build
+import android.util.Base64
+import com.stripe.android.link.LinkConfiguration
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.SetupIntent
+import com.stripe.android.model.StripeIntent
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+
+@Serializable
+internal data class PopupPayload(
+    @SerialName("publishableKey")
+    val publishableKey: String,
+
+    @SerialName("stripeAccount")
+    val stripeAccount: String?,
+
+    @SerialName("merchantInfo")
+    val merchantInfo: MerchantInfo,
+
+    @SerialName("customerInfo")
+    val customerInfo: CustomerInfo,
+
+    @SerialName("paymentInfo")
+    val paymentInfo: PaymentInfo?,
+
+    @SerialName("returnUrl")
+    val returnUrl: String,
+
+    @SerialName("locale")
+    val locale: String,
+) {
+    @SerialName("path")
+    val path: String = "mobile_pay"
+
+    @SerialName("integrationType")
+    val integrationType: String = "mobile"
+
+    @Serializable
+    data class MerchantInfo(
+        @SerialName("businessName")
+        val businessName: String,
+
+        @SerialName("country")
+        val country: String?,
+    )
+
+    @Serializable
+    data class CustomerInfo(
+        @SerialName("email")
+        val email: String?,
+
+        @SerialName("country")
+        val country: String?,
+    )
+
+    @Serializable
+    data class PaymentInfo(
+        @SerialName("currency")
+        val currency: String,
+
+        @SerialName("amount")
+        val amount: Long,
+    )
+
+    fun toUrl(): String {
+        val json = Json.encodeToString(serializer(), this)
+        return baseUrl + Base64.encodeToString(json.encodeToByteArray(), Base64.NO_WRAP)
+    }
+
+    companion object {
+        @Volatile
+        internal var baseUrl: String = "https://checkout.link.com/link-popup.html#"
```

Comment:
Does this need to be var?

Reply:
It doesn't in this PR, but it will in the future (we'll change it for testing the popup in UI tests). But I'll make it private val for now.